packages/account/src/predicate/predicate.ts

```diff
@@ -147,6 +149,46 @@ export class Predicate<
     return mainFn?.encodeArguments(this.predicateData) || new Uint8Array();
   }
 
+  /**
+   * Creates a new Predicate instance from an existing Predicate instance.
+   * @param instance - The existing Predicate instance.
+   * @returns A new Predicate instance with the same bytecode, ABI and provider but with the ability to set the data and configurable constants.
+   */
+  static fromInstance<
+    TData extends InputValue[] = InputValue[],
+    TConfigurables extends { [name: string]: unknown } | undefined = { [name: string]: unknown },
+  >(instance: Predicate<TData, TConfigurables>) {
+    return new (class {
+      data: TData;
+      configurableConstants: TConfigurables;
+
+      constructor() {
+        this.data = instance.predicateData;
+        this.configurableConstants = {} as TConfigurables;
+      }
+
+      withData(data: TData): this {
+        this.data = data;
+        return this;
+      }
+
+      withConfigurableConstants(configurableConstants: TConfigurables): this {
+        this.configurableConstants = configurableConstants;
+        return this;
+      }
+
+      build(): Predicate<TData, TConfigurables> {
+        return new Predicate({
+          bytecode: instance.initialBytecode,
+          abi: instance.interface.jsonAbi,
+          provider: instance.provider,
+          data: this.data,
+          configurableConstants: this.configurableConstants,
+        });
+      }
+    })();
+  }
```

Comment:
```suggestion
  static fromInstance<
    TData extends InputValue[] = InputValue[],
    TConfigurables extends { [name: string]: unknown } | undefined = { [name: string]: unknown },
  >(instance: Predicate<TData, TConfigurables>, overrides?: PredicateParams = {}) {
    return new Predicate({
        bytecode: instance.initialBytecode,
        abi: instance.interface.abi,
        provider: instance.provider,
        data: instance.data,
        configurableConstants: instance.configurableConstants,
        ...overrides,
    });
  }
```

Would the following not be simpler? We'd need to keep a reference to the initial `configurableConstants` with this example, but this could be omitted.

Reply:
I was thinking about something like this, but decide not to store constants. So you think that way is better?

Reply:
I'd be in favour of this route, but the team may have differing opinions.

Reply:
@arboleya @Torres-ssf @danielbate will there be any other comments?

Reply:
Maintaining different APIs around predicate instantiation doesn't seem right to me, should be fluid from one to the next, the builder feels unnecessary IMO. I'm in favour of this change.

Additionally, does this function need to be static? Then we could avoid having to pass `predicate` as first param? 🤔

Reply:
Maintaining different APIs around predicate instantiation doesn't seem right to me, should be fluid from one to the next, the builder feels unnecessary IMO. I'm in favour of this change.

Additionally, does this function need to be static? Then we could avoid having to pass `predicate` as first param? 🤔

Reply:
But in case if we have way to override any of fields via PredicateParams then for me there is no reason to have this method. Predicate can be created with constructor

Reply:
The reason why I decided to make a builder pattern here is to encapsulate logic of overriding only constants and data while remain other fields unchangeable

Reply:
what do you think? @danielbate @petertonysmith94

Reply:
You could pick out the fields that are required to override:

```ts
overrides?: Pick<PredicateParams, 'data' | 'configurableConstants'> = {}
```

Reply:
okay, that fits better. I will implement like this

Reply:
@YaTut1901 resolving this one in favour of the non-static method displayed [here](https://github.com/FuelLabs/fuels-ts/pull/3432#discussion_r1867729684).